### PR TITLE
Fix AdaptiveMC if first or second run fails

### LINKED ISSUE #2027: [DEFECT] Adaptive MonteCarlo crashes if failure occurs at the begin of an evaluation
### Thank you for the defect report

- [X] I am using the latest version of `RAVEN`.
- [X] I have read the [Wiki](https://github.com/idaholab/raven/wiki).
- [x] I have created a [minimum, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)
that demonstrates the defect.

### Defect Description

If a sampling fails in the first iterations, the adaptive Montecarlo crashes when a "successful" model evaluation is achieved. This is due to the fact that the convergence is checked on ```nan``

### Steps to Reproduce

create a model that fails the first run and the code will crash.

### Expected Behavior

since it is a "Montecarlo" it should neglect the failure and go on.

### Screenshots and Input Files

_No response_

### OS

MacOS

### OS Version

_No response_

### Dependency Manager

CONDA

### For Change Control Board: Issue Review

- [X] Is it tagged with a type: defect or task?
- [X] Is it tagged with a priority: critical, normal or minor?
- [X] If it will impact requirements or requirements tests, is it tagged with requirements?
- [X] If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [X] Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

### For Change Control Board: Issue Closure

- [x] If the issue is a defect, is the defect fixed?
- [x] If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by aalfonsi**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #2027

##### What are the significant changes in functionality due to this change request?
Convergence check in Adaptive MC has been changed in order to be able to detect NaN (if not enough data have been collected yet). This disallow the crash of the code in case the first run (evaluation) fails.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: tests/framework/Samplers/AdaptiveMonteCarlo/tests [hunk 1]
```diff
@@ -1,7 +1,12 @@
 [Tests]
- [./test_adaptive_monte_carlo]
+ [./AdaptiveMonteCarlo]
```
Comment (aalfonsi) [inline tests/framework/Samplers/AdaptiveMonteCarlo/tests:2]: In the test file this test is called ```AdaptiveMonteCarlo``` and not ```test_adaptive_monte_carlo```

File: rook/OrderedCSVDiffer.py [hunk 2]
```diff
@@ -192,6 +194,7 @@ def diff(self):
             diffs.append(diff)
             if not match:
               same = False
+              print(match_is_number)
```
Comment (wangcj05) [inline rook/OrderedCSVDiffer.py:197]: remove?

File: tests/framework/Samplers/AdaptiveMonteCarlo/test_adaptive_MC_sampler_with_failures.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      </variable>
+      <variable name="x2">
+        <distribution>x2_dst</distribution>
+      </variable>
+    </AdaptiveMonteCarlo>
+  </Samplers>
+
+  <Models>
+    <ExternalModel ModuleToLoad="adaptive_test_model_with_failures" name="testFunction" subType="">
+      <variables>x1,x2,y1,y2</variables>
```
Comment (wangcj05) [inline tests/framework/Samplers/AdaptiveMonteCarlo/test_adaptive_MC_sampler_with_failures.xml:84]: please use 'inputs' and 'outputs' here

**Review by wangcj05 — CHANGES_REQUESTED** [review]
some general comments for you to consider. @aalfonsi

File: rook/OrderedCSVDiffer.py (hunk 2 above)
Comment (aalfonsi) [reply rook/OrderedCSVDiffer.py]: removed

File: tests/framework/Samplers/AdaptiveMonteCarlo/test_adaptive_MC_sampler_with_failures.xml (hunk 3 above)
Comment (aalfonsi) [reply tests/framework/Samplers/AdaptiveMonteCarlo/test_adaptive_MC_sampler_with_failures.xml]: ops. right. Fixed

**Comment by aalfonsi** [comment]
@wangcj05 addressed comments.

**Review by wangcj05 — APPROVED** [review]

**Comment by wangcj05** [comment]
PR checklist is satisfied, and tests are green. PR can be merged.

**Comment by wangcj05** [comment]
@aalfonsi Thanks for your contribution, I have merged your PR.

**Pull request merged**